### DISCUSSION TITLE
FATAL miktex-epstopdf - Input file name not allowed in restricted mode

### DISCUSSION THREAD

**Discussion by grosenberg** [MiKTeX/miktex#760] [Q&A]
Trying to generate a Pdf from markdown source using Pandoc/MikTex (Windows 7 & basic-miktex-21.2-x64). The markdown includes a latex block to include an EPS-encoded graphic file.

The problem is, miktex-epstopdf is failing with the error "Input file name not allowed in restricted mode". Any insights on what is going on and how to fix/workaround the problem would be greatly appreciated.

The actual latex used to include the EPS-encoded graphic file is:

```
    \begin{figure}[htp]
         \begin{center}
               \graphicspath{{C:/Users/me/AppData/Local/Temp/liveview/pdf-0626/}}
               \includegraphics[width=0.8\\textwidth]{fluent-64543343.eps}
         \end{center}
    \end{figure}
```

The file is verified to exist, is readable by all, and contains the correct content. Note that the file is exists under my Windows standard temporary folder root.

The exact error is:

```
2021-03-21 22:24:08,141-0700 INFO  miktex-epstopdf - this process (9680) started by 'cmd' with command line: miktex-epstopdf --outfile=C:/Users/me/AppData/Local/Temp/liveview/pdf-0626/fluent-64543343-eps-converted-to.pdf C:/Users/me/AppData/Local/Temp/liveview/pdf-0626/fluent-64543343.eps
2021-03-21 22:24:08,150-0700 WARN  miktex-epstopdf - security risk: running with elevated privileges
2021-03-21 22:24:08,290-0700 FATAL miktex-epstopdf - Input file name not allowed in restricted mode: C:/Users/me/AppData/Local/Temp/liveview/pdf-0626/fluent-64543343.eps
2021-03-21 22:24:08,290-0700 FATAL miktex-epstopdf - Input file name not allowed in restricted mode: C:/Users/me/AppData/Local/Temp/liveview/pdf-0626/fluent-64543343.eps
```

ReportDate: 2021-03-21 23:10:25
CurrentVersion: 21.3
SetupDate: 2021-03-21 20:52:00
SetupVersion: 21.2
Configuration: Regular
GitInfo: ab8ebca / 2021-03-19 10:22:46
OS: Windows 7 Home Premium, 64-bit, Service Pack 1, build 7601
SharedSetup: yes
LinkTargetDirectory: C:\Program Files\MiKTeX\miktex\bin\x64
PathOkay: yes
LastUpdateCheckAdmin: 2021-03-21 20:55:20
LastUpdateAdmin: 2021-03-21 20:53:53
LastUpdateDbAdmin: 2021-03-21 20:55:19
SystemAdmin: yes
RootPrivileges: yes
AdminMode: yes
Root0: C:\ProgramData\MiKTeX
Root1: C:\Program Files\MiKTeX
CommonInstall: C:\Program Files\MiKTeX
CommonConfig: C:\ProgramData\MiKTeX
CommonData: C:\ProgramData\MiKTeX

**Comment by edocevoli** [comment]
MiKTeX now uses the restricted version of `epstopdf`:  see `epstopdf-sys.cfg`.
In restricted mode, absolute path names are not allowed for security reasons.

**Comment by grosenberg** [comment]
Thanks for the prompt response.

Is using the system temp folder structure completely precluded? Creating lots of temporary eps files in the document local user file space seems to be a *really* bad idea.

Tried adding `\def\pdfshellescape{1}` to the latex block to no effect.

Thanks.

**Comment by grosenberg** [comment]
Tried using a document local temporary directory to contain the generated eps files: 

`pdf-0156/fluent-51647338.eps` -- works (but is visible to the user)
`.pdf-0156/fluent-51647338.eps` -- fails with `Input file name not allowed in restricted mode `
`./.pdf-0156/fluent-51647338.eps` -- fails with `Input file name not allowed in restricted mode `

Any suggestions on a workable solution would be appreciated.

**Comment by edocevoli** [comment]
You have to turn on "unrestricted" mode if you have to use unsafe file names: `--shell-escape`

**Reply by mmtt27** [reply]
I have the same error and same question.  Where does one input the "--shell-escape" command?

**Reply by alex-fdias** [reply]
I guess you most likely found an answer in the meantime, in any case and for those like me who came across your question, in TeXworks/Windows this can be configured by going to "Edit -> Preferences", choosing the "Processing tool" from the list (double click or press "Edit...") and, for example, for pdfLaTeX you can click the "+" button to add an argument and type or paste "--shell-escape". Double-clicking an argument allows editing that argument.

**Comment by grosenberg** [comment]
Very much appreciate the hint.

For the Pandoc command line, the hint lead to using

                  --pdf-engine=pdflatex --pdf-engine-opt=--shell-escape

JFYI, the MikTeX manual makes no (easily findable) mention of "unrestricted mode" and the [PdfLatex Doc page](https://docs.miktex.org/manual/miktex-pdftex.html) does not list `--shell-escape` as an option.

Again, thanks for attending to my question.

**Comment by skhiggins** [comment]
I had the same issue but I am using a **relative** file path. Are relative file paths also not allowed in restricted mode?

Output:
```
2021-10-21 10:35:07,329-0500 INFO  miktex-epstopdf - this process (7544) started by 'cmd' with command line: miktex-epstopdf --outfile=../graphs/branch_dep_with_w5_control_cropped-eps-converted-to.pdf ../graphs/branch_dep_with_w5_control_cropped.eps
2021-10-21 10:35:07,520-0500 FATAL miktex-epstopdf - Input file name not allowed in restricted mode: ../graphs/branch_dep_with_w5_control_cropped.eps
```
